SimTracker/TrackTriggerAssociation/plugins/StubAssociator.cc

```diff
@@ -0,0 +1,114 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/EDPutToken.h"
+#include "DataFormats/Common/interface/Handle.h"
+
+#include "SimTracker/TrackTriggerAssociation/interface/TTTypes.h"
+#include "SimTracker/TrackTriggerAssociation/interface/StubAssociation.h"
+#include "L1Trigger/TrackTrigger/interface/Setup.h"
+
+#include <vector>
+#include <map>
+#include <utility>
+#include <set>
+#include <algorithm>
+#include <iterator>
+#include <cmath>
+
+using namespace std;
+using namespace edm;
+
+namespace tt {
+
+  /*! \class  tt::StubAssociator
+   *  \brief  Class to associate reconstrucable TrackingParticles with TTStubs and vice versa.
+   *          It may associate multiple TPs with a TTStub and can therefore be used to associate
+   *          TTTracks with TrackingParticles.
```

Comment:
This comment should make clear that "This EDProducer creates two StubAssociation EDProducts, one using only TP that are "reconstructable" (produce stubs in a min. number of layers) & one using TP that are also "use for the tracking efficiency measurement" (also pass kinematic requirements).

Reply:
done